stripe.go

```diff
@@ -351,13 +351,21 @@ func (s *BackendImplementation) Do(req *http.Request, body *bytes.Buffer, v inte
 			break
 		}
 
-		resBody, err := ioutil.ReadAll(res.Body)
-		res.Body.Close()
-		if err != nil {
-			if s.LogLevel > 0 {
-				s.Logger.Printf("Cannot read response: %v\n", err)
+		var (
+			resBody []byte
+			err     error
+		)
+		// Checking res instead of err because s.HTTPClient.Do(req) has one special case for Go 1 compatibility
+		// when both the response and an error are returned
```

Comment:
Can we clarify this comment a bit? It should read something like "when both the response and error are returned as nil" I think?

Also, what exactly do you mean by "Go 1 compatibility"? All Go versions right now are Go 1 right? Do you mean Go 1.10 or something like that?

Reply:
My comment was based on the following statement made in spripe-go's dependency `go-1.10/src/net/http/client.go` (lines 602 - 605) : 
```
				// Special case for Go 1 compatibility: return both the response
				// and an error if the CheckRedirect function failed.
				// See https://golang.org/issue/3795
				// The resp.Body has already been closed.
```
Current github repo don't have this comment anymore. Anyway, since I understood this library belongs to Go's core, I assumed that must make sense and didn't give it more thought. I still don't know what exactly they meant, so I updated my comment.